databricks/koalas/tests/test_frame_plot.py

```diff
@@ -278,81 +311,115 @@ def test_pie_plot_error_message(self):
         self.assertTrue(error_message in str(context.exception))
 
     def test_scatter_plot(self):
-
-        def _test_scatter_plot(pdf, kdf, x, y, c):
+        def check_scatter_plot(pdf, kdf, x, y, c):
             ax1 = pdf.plot.scatter(x=x, y=y)
+            bin1 = self.plot_to_base64(ax1)
             ax2 = kdf.plot.scatter(x=x, y=y)
-            self.compare_plots(ax1, ax2)
+            bin2 = self.plot_to_base64(ax2)
+            self.assertEqual(bin1, bin2)
 
             ax1 = pdf.plot(kind='scatter', x=x, y=y)
+            bin1 = self.plot_to_base64(ax1)
             ax2 = kdf.plot(kind='scatter', x=x, y=y)
-            self.compare_plots(ax1, ax2)
+            bin2 = self.plot_to_base64(ax2)
+            self.assertEqual(bin1, bin2)
 
             # check when keyword c is given as name of a column
             ax1 = pdf.plot.scatter(x=x, y=y, c=c, s=50)
+            bin1 = self.plot_to_base64(ax1)
             ax2 = kdf.plot.scatter(x=x, y=y, c=c, s=50)
-            self.compare_plots(ax1, ax2)
+            bin2 = self.plot_to_base64(ax2)
+            self.assertEqual(bin1, bin2)
 
         # Use pandas scatter plot example
-        pdf = pd.DataFrame(np.random.rand(50, 4), columns=['a', 'b', 'c', 'd'])
-        kdf = koalas.from_pandas(pdf)
-        _test_scatter_plot(pdf, kdf, x='a', y='b', c='c')
+        pdf1 = pd.DataFrame(np.random.rand(50, 4), columns=['a', 'b', 'c', 'd'])
+        kdf1 = koalas.from_pandas(pdf1)
+        check_scatter_plot(pdf1, kdf1, x='a', y='b', c='c')
 
         # multi-index columns
         columns = pd.MultiIndex.from_tuples([('x', 'a'), ('x', 'b'), ('y', 'c'), ('z', 'd')])
-        pdf.columns = columns
-        kdf.columns = columns
-        _test_scatter_plot(pdf, kdf, x=('x', 'a'), y=('x', 'b'), c=('y', 'c'))
+        pdf1.columns = columns
+        kdf1.columns = columns
+        check_scatter_plot(pdf1, kdf1, x=('x', 'a'), y=('x', 'b'), c=('y', 'c'))
 
     def test_hist_plot(self):
-
-        def _test_hist_plot(pdf, kdf):
+        def check_hist_plot(pdf, kdf):
             _, ax1 = plt.subplots(1, 1)
```

Comment:
qq: some tests have lines like this, but what's those for? Do we need those?

Reply:
I think it was just to test with subplot (to allocate multiple plots (?)). We should close it to properly compare in this case too anyway.. should be fine.